### PR TITLE
fix: set active index correctly when default value is passed in

### PR DESCRIPTION
**Pull request by atomictangerine**
## What I did

Fix the defaultValue bug in tabs component

### CODE REVIEW

File: ui/tabs/src/context.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import * as React from "react";
+
+interface TabsContextInterface {
+  defaultValue?: string;
```
Comment (wp-aberg) [inline ui/tabs/src/context.ts:4]: What if instead of separate values, there is a single `initialValue` ?

File: ui/tabs/src/TabsRoot.tsx [hunk 2]
```diff
@@ -19,9 +20,16 @@ export type TabsRootProps = {
 export const TabsRoot = React.forwardRef<HTMLDivElement, TabsRootProps>(
   ({ children, defaultValue = "", ...props }: TabsRootProps, ref) => {
     return (
-      <StyledTabsRoot defaultValue={defaultValue} {...props} ref={ref}>
-        {children}
-      </StyledTabsRoot>
+      <TabsContext.Provider
+        value={{
+          defaultValue: defaultValue,
```
Comment (wp-aberg) [inline ui/tabs/src/TabsRoot.tsx:25]: and then here `initialValue: defaultValue || props.value`

**Comment by wp-aberg** [comment]
I think adding this to TabsTrigger.tsx will fix the missing animation in the default example

```javascript
React.useEffect(() => {
  if (active && !previousRect) {
    if (typeof setPreviousRect === "function" && internalRef.current) {
      setPreviousRect(internalRef.current.getBoundingClientRect());
    }
  }
}, []);
```

File: ui/tabs/src/context.ts (hunk 1 above)
Comment (atomictangerine) [reply ui/tabs/src/context.ts]: good idea

**Review by artmsilva — APPROVED** [review]

**Pull request merged**